packages/network/lib/connect.ts

```diff
@@ -90,6 +90,12 @@ export function createRetryingSocket (
         return cb(err)
       }
 
+      if (delay === -1) {
+        debug('cancelling any additional retries %o', { opts })
+
+        return cb(err)
+      }
```

Comment:
isn't this what the code path above already does, except with `undefined` instead of `-1`?

Reply:
My initial intention was to differentiate between cancelling early and exhausting the retries but on second thought, I think you're right, they really are the same. Updated [037fe44](https://github.com/cypress-io/cypress/pull/24155/commits/037fe444520a1ba0930046c8fd729230d2ec7684)